app/src/main/java/com/android/sample/ui/account/AccountScreen.kt

```diff
@@ -163,6 +175,12 @@ private fun ListSelection(
           if (selectedListIndex == 0) {
             TopCornerUnLikeButton(recipe, userViewModel)
           } else {
+            TopCornerEditButton(recipe) { selectedRecipe ->
+              createRecipeViewModel.resetInitializationState()
+              createRecipeViewModel.initializeRecipeForEditing(selectedRecipe)
+              navigationActions.navigateTo(Screen.EDIT_RECIPE)
+              Modifier.testTag(RECIPE_EDIT_ICON_TEST_TAG)
```

Comment:
I had multiple problems, so i had to refactor a lot of time. I was porbably not very focus as i refator those method for too much time. So this will be fixed and the `recipeBuilder.clear()`will be called by `resetInitializationState()`